app/org/sagebionetworks/bridge/services/HealthCodeServiceImpl.java

```diff
@@ -43,15 +44,43 @@ public String getCode() {
     }
 
     @Override
-    public String getHealthCode(String id) {
-        return healthIdDao.getCode(id);
+    public HealthId getMapping(String healthId) {
+        checkNotNull(healthId);
+        final String healthCode = healthIdDao.getCode(healthId);
+        if (healthCode == null) {
+            return null;
+        }
+        return getHealthIdObject(healthId, healthCode);
     }
-
+    
     @Override
-    public String getStudyIdentifier(String healthCode) {
-        return healthCodeDao.getStudyIdentifier(healthCode);
+    public HealthId getMapping(Account account) {
+        checkNotNull(account);
+        
+        final String healthId = account.getHealthId();
+        if (healthId == null) {
+            return null;
+        }
```

Comment:
[POLISH] Is it possible to have an account with no health ID? Should we have a unit test to test this scenario?